File: bindings/java/src/main/java/io/uds/DeviceInfo.java
package io.uds;

import java.util.*;
import com.google.gson.*;
import com.google.gson.reflect.*;

public class DeviceInfo {
    public String id;
    public String name;
    public String transport;
    public Boolean connected;
    public String firmwareVersion;
    public String platform;
    public Map<String, Object> capabilities;
    public Long uptimeSeconds;

    public static List<DeviceInfo> fromJsonArray(String json) {
        Gson gson = new Gson();
        Type listType = new TypeToken<List<DeviceInfo>>(){}.getType();
        return gson.fromJson(json, listType);
    }

    public static DeviceInfo fromJson(String json) {
        return new Gson().fromJson(json, DeviceInfo.class);
    }

    public String toJson() {
        return new Gson().toJson(this);
    }

    @Override
    public String toString() {
        return String.format("DeviceInfo{id='%s', name='%s', connected=%s, platform='%s'}",
            id, name, connected, platform);
    }
}

File: bindings/kotlin/src/main/kotlin/io/uds/DeviceInfo.kt
package io.uds

import com.google.gson.annotations.SerializedName

data class DeviceInfo(
    val id: String,
    val name: String? = null,
    val transport: String? = null,
    val connected: Boolean = false,
    @SerializedName("firmware_version")
    val firmwareVersion: String? = null,
    val platform: String? = null,
    val capabilities: Map<String, Any>? = null,
    @SerializedName("uptime_seconds")
    val uptimeSeconds: Long? = null
)
